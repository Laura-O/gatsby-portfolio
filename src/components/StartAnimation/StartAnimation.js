import React from 'react';
import styles from './styles.module.scss';

export default function StartAnimation() {
  return (
    <div className="hello">
      <svg viewBox="0 0 600 300">
        <symbol id="s-text">
          <text textAnchor="left" x="15%" y="25%" dy=".45em">
            Hi!
          </text>
        </symbol>

        <use xlinkHref="#s-text" className={styles.text} />
        <use xlinkHref="#s-text" className={styles.text} />
        <use xlinkHref="#s-text" className={styles.text} />
        <use xlinkHref="#s-text" className={styles.text} />
        <use xlinkHref="#s-text" className={styles.text} />
      </svg>
    </div>
  );
}
