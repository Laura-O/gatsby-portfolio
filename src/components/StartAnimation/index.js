import React, { Component } from 'react';
import Link from 'gatsby-link';

import styles from './styles.module.scss';

class StartAnimation extends Component {
  render() {
    return (
      <div>
        <svg viewBox="0 0 600 300">
          <symbol id="s-text">
            <text textAnchor="middle" x="50%" y="50%" dy=".35em">
              Hi!
            </text>
          </symbol>

          <use xlinkHref="#s-text" className={styles.text} />
          <use xlinkHref="#s-text" className={styles.text} />
          <use xlinkHref="#s-text" className={styles.text} />
          <use xlinkHref="#s-text" className={styles.text} />
          <use xlinkHref="#s-text" className={styles.text} />
        </svg>
      </div>
    );
  }
}

export default StartAnimation;
