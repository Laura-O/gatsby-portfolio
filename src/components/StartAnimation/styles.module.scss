@import '../../scss/variables.scss';

$colors: #fff, $redPurple, $yellow, $linkColor, $malachite;

$max: 4;
$stroke-step: 6%;
$stroke-space: 1%;
$stroke-time: 15s;
$stroke-time-step: $stroke-time / $max;

.text {
  fill: none;
  stroke-width: 4;
  stroke-linejoin: round;
  stroke-dasharray: $stroke-step
    ($stroke-step * ($max - 1) + $stroke-space * $max);
  stroke-dashoffset: 0;
  -webkit-animation: stroke $stroke-time infinite linear;
  animation: stroke $stroke-time infinite linear;
  font-size: 10em;

  @for $item from 1 through $max {
    $stroke-color: nth($colors, $item);

    &:nth-child(5n + #{$item}) {
      stroke: $stroke-color;
      animation-delay: -$stroke-time-step * $item;
    }
  }

  &:hover {
    background-color: white;
  }
}

@-webkit-keyframes stroke {
  100% {
    stroke-dashoffset: -400;
  }
}

@keyframes stroke {
  100% {
    stroke-dashoffset: -400;
  }
}

svg {
  position: absolute;
  width: 100%;
  height: 30%;
}
