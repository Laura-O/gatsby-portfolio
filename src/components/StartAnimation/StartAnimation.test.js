import React from 'react';
import { shallow } from 'enzyme';
import StartAnimation from './StartAnimation';

test('rendering', () => {
  const wrapper = shallow(<StartAnimation path="/something" />);
  expect(wrapper).toMatchSnapshot();
});
