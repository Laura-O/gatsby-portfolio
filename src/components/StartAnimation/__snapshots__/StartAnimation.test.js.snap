// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering 1`] = `
<div
  className="hello"
>
  <svg
    viewBox="0 0 600 300"
  >
    <symbol
      id="s-text"
    >
      <text
        dy=".45em"
        textAnchor="left"
        x="15%"
        y="25%"
      >
        Hi!
      </text>
    </symbol>
    <use
      xlinkHref="#s-text"
    />
    <use
      xlinkHref="#s-text"
    />
    <use
      xlinkHref="#s-text"
    />
    <use
      xlinkHref="#s-text"
    />
    <use
      xlinkHref="#s-text"
    />
  </svg>
</div>
`;
